src/integrationTest/java/org/opensearch/test/framework/cluster/TestRestClient.java

```diff
@@ -284,7 +286,26 @@ public HttpResponse(CloseableHttpResponse inner) throws IllegalStateException, I
             this.header = inner.getHeaders();
             this.statusCode = inner.getCode();
             this.statusReason = inner.getReasonPhrase();
+
             inner.close();
+
+            if (this.body.length() != 0) {
+                verifyContentType();
+            }
+        }
+
+        private void verifyContentType() {
```

Comment:
It is lame that I needed to duplicate this code, but it didn't seem right to move this logic into the plugin code where it would be reusable, or to spin up a whole new shared module... yet.